pkg/crc/cluster/cluster.go

```diff
@@ -178,21 +178,55 @@ Environment=NO_PROXY=%s,%s`
 	return nil
 }
 
-// Only marketplace operator doesn't get updated with proxy when it is set using proxy resource
-// All other operators which have `config.openshift.io/inject-proxy` annotation get updated except marketplace.
-func AddProxyConfigToMarketplaceOperator(oc oc.OcConfig, proxy *network.ProxyConfig) error {
-	cmdArgs := []string{"set", "env", "deployment", "marketplace-operator", "-n", "openshift-marketplace",
+func addProxyConfigToDeployment(oc oc.OcConfig, deployment string, namespace string, proxy *network.ProxyConfig) error {
+	logging.Debugf("Adding proxy configuration to %s/%s", namespace, deployment)
+	cmdArgs := []string{"set", "env", "deployment", deployment, "-n", namespace,
 		fmt.Sprintf("HTTP_PROXY=%s", proxy.HttpProxy),
 		fmt.Sprintf("HTTPS_PROXY=%s", proxy.HttpsProxy),
 		fmt.Sprintf("NO_PROXY=%s,%s", proxy.GetNoProxyString(), internalNoProxy),
 	}
+	if _, stderr, err := oc.RunOcCommand(cmdArgs...); err != nil {
+		return fmt.Errorf("Failed to add proxy details to %s (namespace: %s) %v: %s", deployment, namespace, err, stderr)
+	}
+	return nil
+}
+
+// crc clusters are not running the cluster-version-operator, which is
+// responsible for adding proxy env variables to deployments which have the
+// 'inject-proxy' annotation.
+// This is why we have to set the proxy configuration manually on some of the
+// deployments
+func addProxyConfigToDeployments(oc oc.OcConfig, proxy *network.ProxyConfig) error {
 	if err := oc.WaitForOpenshiftResource("deployment"); err != nil {
 		return err
 	}
-	if _, stderr, err := oc.RunOcCommand(cmdArgs...); err != nil {
-		return fmt.Errorf("Failed to add proxy details to marketplace-operator %v: %s", err, stderr)
+
+	multiErr := errors.MultiError{}
+	err := addProxyConfigToDeployment(oc, "marketplace-operator", "openshift-marketplace", proxy)
+	if err != nil {
+		logging.Debugf("Failed to add proxy settings to marketplace operator: %v", err)
+		multiErr.Collect(err)
 	}
-	return nil
+
+	err = addProxyConfigToDeployment(oc, "ingress-operator", "openshift-ingress-operator", proxy)
+	if err != nil {
+		logging.Debugf("Failed to add proxy settings to ingress operator: %v", err)
+		multiErr.Collect(err)
+	}
+
+	err = addProxyConfigToDeployment(oc, "cluster-image-registry-operator", "openshift-image-registry", proxy)
+	if err != nil {
+		logging.Debugf("Failed to add proxy settings to image registry operator: %v", err)
+		multiErr.Collect(err)
+	}
+
+	err = addProxyConfigToDeployment(oc, "openshift-controller-manager-operator", "openshift-controller-manager-operator", proxy)
+	if err != nil {
+		logging.Debugf("Failed to add proxy settings to image registry operator: %v", err)
+		multiErr.Collect(err)
+	}
+
+	return multiErr.ToError()
 }
```

Comment:
Was just wondering if you had considered using a map of strings to store the deployment name and namespace then looping over it to add the proxy config instead of duplicating?

Reply:
we can do that but I think if @cfergeau test around the bundle which have CVO not disabled make all those obsolete then we might not even need this one.